### ISSUE TITLE
Set time bug

### ISSUE THREAD

**Issue by toreskogp** [spencermountain/spacetime#63]
Having a problem setting the time using immutable. Trying to set the time to 08:00, but for some reason it seems to be set to the current hour.

I reproduced the bug with runkit: https://runkit.com/toreskog/5ba7999d0612d500122fbb97

```
const spacetime = require("spacetime/immutable")

const day = spacetime('2018-09-23').time('06:00')
console.log(day.format('iso')) // --> "2018-09-23T15:00:00.000+02:00"
```

**Comment by spencermountain** [comment]
hey @toreskogp whoa, this is weird -  i [cloned](https://runkit.com/spencermountain/5ba8130a6e721c0011359e48) your example, did nothing else, and it worked....
yours:
![image](https://user-images.githubusercontent.com/399657/45933666-3bbf6f00-bf5f-11e8-975f-cf19bd780ccb.png)

mine:
![image](https://user-images.githubusercontent.com/399657/45933664-33ffca80-bf5f-11e8-88b6-186489aab0d5.png)

now if that isn't a fun-looking bug, I don't know what is.

my only guess is that runkit does caching, and this is a time-sensitive bug. That would make some sense, that the bug is only reproducible at a certain time of day. I've seen that before.

yikes!
have a busy week, but will take a look when I have some time.
lemme know if you find any clues.
cheers

**Comment by toreskogp** [comment]
Yes, it seems like runkit is caching the result. But I can rerun it and the bug is there every time. I have tried yesterday at between 15:00 and 16:00, 16:00 and 17:00 and 21:00 and 22:00.

It seems like no matter what time I try to set the time ends up being the beginning of the current hour. And it has been perfectly reproducible.
I runned it now at 11.23 and the time ended up being 11:00.

And also it seems like you have the bug as well. The time is tried to set to 6:00 in the morning and the result is 18:00 in the afternoon. So chanses are the you tried to run the script between 18:00 and 19:00?

And thanks for looking in to this. I really like your library 😉

**Comment by spencermountain** [comment]
ha! yeah thanks. happy to look at this in a few days.

**Comment by spencermountain** [comment]
hey, i've got this fixed on master, not gonna release it I don't think, cause it's friday-night, and it's a holiday weekend (canadian thanksgiving)

this was a good find!
thank you

**Comment by spencermountain** [comment]
shipped in `4.5.0`

**Comment by toreskogp** [comment]
Thanks a lot!!!
